.github/workflows/release-bundle.yaml

```diff
@@ -0,0 +1,35 @@
+name: Release bundle to CharmHub
+on:
+  workflow_call:
+    inputs:
+      release:
+        description: Bundle to run release to Charmhub (i.e. 1.10/candidate)
+        type: string
+        required: true
+    secrets:
+      CHARMCRAFT_CREDENTIALS:
+        required: true
+  workflow_dispatch:
+    inputs:
+      release:
+        description: Bundle to run release to Charmhub (i.e. 1.10/candidate)
+        type: string
+        required: true
+    secrets:
+      CHARMCRAFT_CREDENTIALS:
+        required: true
```

Comment:
question: what about on push? the old workflow was releasing the bundle on push. I don't think we should release without testing the bundle, but at the same time don't want to add more manual processes.

Reply:
Hmm that's a good question!

Initially I was thinking that this would only be manual, for the reasons you also suggested (not pushing something we haven't tested). But thinking about it, I'm actually OK if we also release to charmhub on push, but only for `edge` track.

@NohaIhab @orfeas-k WDYT?

Reply:
I think it's fine to have this on push actually, since in most cases we merge this after we have tested e.g. candidate and then promoted everything to stable.

Reply:
releasing to edge makes sense, I would expect any changes made to stable bundle would be done to edge bundle too. Do you think there's a case where we don't apply the same changes to edge?
Then we can gate the stable release by running the tests, getting feedback from SolQA and so on.

Reply:
> since in most cases we merge this after we have tested e.g. candidate and then promoted everything to stable.

it's true that's how we've done it in the past. On second thought, it's fair to expect whichever bundle you made changes to, to be released.

I would prefer if the tests were integrated in the CI but I don't want to block this PR any further.
The tests were being skipped anyway. Can we have an issue to integrate the tests on push? It can even be done when migrating to the terraform solutions.

Reply:
> Then we can gate the stable release by running the tests, getting feedback from SolQA and so on.

As long as SQA is gating a bundle's release, I don't see a way to merge a PR -> SQA tests the bundle.yaml -> bundle is released automatically. I think it's fair to assume that for the actual release process, we 're testing candidate bundles and once everything's good, then we can just promote all charms to `stable` risk and also publish a `stable` bundle. I think it's also what we have defined for stable release in the SQA testing https://docs.google.com/document/d/1ZJGeHfC2AXPBhfqdbPwa8OWhjk5haNxi70165pw4Xj8/edit?tab=t.0#heading=h.yo8iigavhyre spec.

Reply:
> I would prefer if the tests were integrated in the CI but I don't want to block this PR any further.

I think it is worth to see if this can be done differently in Terraform, as you pointed out Noha since bundle are being deprecated.

Reply:
Created https://github.com/canonical/bundle-kubeflow/issues/1322 to track the next steps